shell/format.sh

```diff
@@ -3,6 +3,10 @@ set -Eeuo pipefail
 
 base_dir=$(dirname $(dirname $0))
 
+echo "Generating public APIs..."
+python3 "${base_dir}"/api_gen.py
```

Comment:
Upon thinking more about it -- an issue with generating the API in this script is that it takes quite a bit of time, but most of the time it won't be needed (it's only needed when we add new APIs). So it's actually better to have it in a separate script after all.

Reply:
Thanks for the feedback. Added `shell/api_gen.sh` to generate APIs and kept `shell/format.sh` for Code formatting.